app-typescript/components/TreeSelect/TreeSelect.tsx

```diff
@@ -671,38 +701,98 @@ export class TreeSelect<T> extends React.Component<IProps<T>, IState<T>> {
                                     <i className="icon-plus-large"></i>
                                 </button>
                             }
+                            {this.props.dragAndDrop
```

Comment:
Reduce code duplication - do not repeat `<TreeSelect />` multiple times. Here's an example:

```
// BAD

if (draggable) {
    <ul className="draggable-wrapper">
        {items.map((item) => (
            <div className="draggable-item">
                <TreeSelect />
            </div>
        ))}
    </ul>
} else {
    <ul className="NON-draggable-wrapper">
        {items.map((item) => (
            <div className="NON-draggable-item">
                <TreeSelect />
            </div>
        ))}
    </ul>
}


// GOOD

const MainWrapper = draggable
    ? (children) => (<ul className="draggable-wrapper">{children}</ul>)
    : (children) => (<ul className="NON-draggable-wrapper">{children}</ul>);

const ItemWrapper = draggable
    ? (children) => (<ul className="draggable-item">{children}</ul>)
    : (children) => (<ul lassName="NON-draggable-item">{children}</ul>);

return (
    <MainWrapper>
        {items.map((item) => (
            <ItemWrapper>
                <TreeSelect />
            </ItemWrapper>
        ))}
    </MainWrapper>
);
```